galsim/wfirst/wfirst_detectors.py

```diff
@@ -83,26 +82,80 @@ def applyIPC(img, edge_treatment='extend', fill_value=None):
 
-def applyPersistence(img, prev_exposures):
+def applyPersistence(img, prev_exposures, method='fermi'):
     """
+    This method applies either of the two different persistence models: 'linear' and 'fermi'.
+    Slew between pointings and consecutive resets after illumination are not considered.
+
+    'linear' persistence model:
     Applies the persistence effect to the Image instance by adding a small fraction of the previous
     exposures (up to {ncoeff}) supplied as the 'prev_exposures' argument.
-
     For more information about persistence, see the docstring for galsim.Image.applyPersistence.
     Unlike that routine, this one does not need the coefficients to be specified. However, the list
-    of previous eight exposures will have to be supplied. Earlier exposures, if supplied, will be
+    of previous {ncoeff} exposures will have to be supplied. Earlier exposures, if supplied, will be
     ignored.
 
+    'fermi' persistence model:
+    Applies the persistence effect to the Image instance by adding the accumulated persistence dark
+    current of previous exposures supplied as the 'prev_exposures' argument.
+    Unlike galsim.Image.applyPersistence, this one does not use constant coefficients but a fermi
+    model plus a linear tail below half of saturation.
+    See http://www.stsci.edu/hst/wfc3/ins_performance/persistence/ for more info about the fermi model.
+
     @param img               The Image to be transformed.
-    @param prev_exposures    List of up to {ncoeff} Image instances in the order of exposures, with the
-                             recent exposure being the first element.
+    @param prev_exposures    List of Image instances in the order of exposures, with the recent
+                             exposure being the first element. In the linear model, the exposures
+                             exceeding the limit ({ncoeff} exposures) will be ignored.
+    @param method            The persistence model ('linear' or 'fermi') to be applied.
+                             [default: 'fermi']
     """.format(ncoeff=len(galsim.wfirst.persistence_coefficients))
+
     if not hasattr(prev_exposures,'__iter__'):
-        raise TypeError("In wfirst.applyPersistence, prev_exposures must be a list of Image "
-                        "instances")
+        raise TypeError("In wfirst.applyPersistence, prev_exposures must be a list of Image instances")
+
+    if method == 'linear':
+
+        n_exp = min(len(prev_exposures),len(galsim.wfirst.persistence_coefficients))
+        img.applyPersistence(prev_exposures[:n_exp],galsim.wfirst.persistence_coefficients[:n_exp])
+
+    elif method == 'fermi':
+
+        n_exp = len(prev_exposures)
+        for i in range(n_exp):
+            ### The length of time where slews and pixel resets takes should be specified.
+            ### Now it is simply assumed to be 0 until we get more information.
+            img += fermi_linear(prev_exposures[i].array)/(1.+ 2*i)
```

Comment:
While this works, it would be slightly clearer imo to write
```
img.array += fermi_linear(...
```
I was actually surprised that image += array was valid, but it's a byproduct of our code to allow image += float in addition to image += image.

Reply:
This line is updated as suggested.

In addition, I actually make an approximation here in this line. Since the persistence current is inversely proportional to time, the integrated persistence signal over the exposure time is
<img width="228" alt="image" src="https://user-images.githubusercontent.com/39343677/58375497-3a957600-7f22-11e9-935f-d8bd777535ec.png">, where t0 is the starting time of a previous exposure since reset and t_exp is the exposure time (140.25 s). The integrated persistence signal is sensitive t0, but the timeline of observation is  currently not available. So I take the persistence current at the mid-time of a exposure as the average persistence current and approximate the integrated persistence signal as 
<img width="205" alt="image" src="https://user-images.githubusercontent.com/39343677/58375553-aa583080-7f23-11e9-8017-b8f8d5dfee2b.png">; i.e., I'm approximating the 1/t curve locally with a straight line. 
![time_integrated](https://user-images.githubusercontent.com/39343677/58375584-3c603900-7f24-11e9-8eac-cbdbf4a7704f.png)
For more distant previous exposures (t0 > 140s), it doesn't make much difference. For the last one previous exposure, t0 (the reset time) is likely to be somewhere between 15-30s. This approximation can only provide a rough estimation for the last exposure. More precise calculation would require the whole timeline of observations.